backend/src/test/java/com/javahelp/backend/domain/user/login/LoginInteractorTest.java

```diff
@@ -0,0 +1,67 @@
+package com.javahelp.backend.domain.user.login;
+
+import static org.junit.Assert.assertTrue;
+
+import com.javahelp.backend.data.ITokenStore;
+import com.javahelp.backend.data.IUserStore;
+import com.javahelp.model.user.ClientUserInfo;
+import com.javahelp.model.user.SHAPasswordHasher;
+import com.javahelp.model.user.User;
+import com.javahelp.model.user.UserPassword;
+
+import org.junit.Test;
+
+/**
+ * Tests for {@link LoginInteractor}
+ */
+public class LoginInteractorTest {
+
+    @Test
+    public void testLogin() {
+        IUserStore db = IUserStore.getDefaultImplementation();
+        ITokenStore tokens = ITokenStore.getDefaultImplementation();
+
+        UserPassword p = new UserPassword("password", SHAPasswordHasher.getInstance());
+
+        ClientUserInfo info = new ClientUserInfo("sdjkfhkls", "sdf", "sdf",
+                "sdfhjkl", "sdjlfhkslkjd");
+
+        User u = new User("", info, "testing_user_login_123");
+
+        db.create(u, p);
+
+        LoginInteractor interactor = new LoginInteractor(db, tokens);
+
+        LoginResult result = interactor.login(new ILoginInput() {
```

Comment:
I'm not too familiar, but is this necessarily a good practice? To implement an interface in a test class?

Reply:
Yeah using an anonymous implementation keeps everything simple!